lnwallet/btcwallet/btcwallet.go

```diff
@@ -861,6 +865,22 @@ func (b *BtcWallet) ImportAccount(name string, accountPubKey *hdkeychain.Extende
 	dryRun bool) (*waddrmgr.AccountProperties, []btcutil.Address,
 	[]btcutil.Address, error) {
 
+	// For custom accounts, we first check if there is no existing account
+	// with the same name.
+	if name != lnwallet.DefaultAccountName &&
+		name != waddrmgr.ImportedAddrAccountName {
+
+		_, err := b.ListAccounts(name, nil)
+		if err == nil {
```

Comment:
I think this would be a slight bit easier to understand if formatted as:
```
		if err == nil || 
			waddrmgr.IsError(err, waddrmgr.ErrAccountNotFound) {

			return nil, nil, nil,
				fmt.Errorf("account '%s' already exists",
					name)
		}
		if err != nil {
			return nil, nil, nil, err
		}
```

Reply:
I do agree, however, IMO returning the error "account already exists" when we receive the error `waddrmgr.ErrAccountNotFound` can be misleading, no?

Reply:
Ah right. Let's leave it the way it is then.